docs/how-to-guides.md

```diff
@@ -32,6 +32,72 @@ Everything about test data generation and its adjustable parameters can
 be found in the [Generating a Test Dataset with niceML](
 generate-data.md#step-3-customizing-data-generation-optional)
 
+## How to add a custom model
+In order to define a custom model in niceML we can make use of the niceML `ModelFactory` class
+and its create_model function. To create a new model, implement yours by inheriting from 
+the `ModelFactory` and configuring the functions you need. Here is an example:
+```python
+from typing import Any
+
+import tensorflow as tf
+from keras import layers, Sequential, regularizers
+from niceml.data.datadescriptions.datadescription import DataDescription
+from niceml.data.datadescriptions.inputdatadescriptions import InputImageDataDescription
+from niceml.data.datadescriptions.outputdatadescriptions import OutputVectorDataDescription
+from niceml.mlcomponents.models.modelfactory import ModelFactory
+from niceml.utilities.commonutils import check_instance
+
+
+class FlowerCNN(ModelFactory):
+    def __init__(self, regulation_value: float, activation_function: str, final_activation: str):
+        self.final_activation = final_activation
+        self.activation_function = activation_function
+        self.regulation_value = regulation_value
+
+    def create_model(self, data_description: DataDescription) -> Any:
+        input_dd: InputImageDataDescription = check_instance(
+            data_description, InputImageDataDescription
+        )
+        output_dd: OutputVectorDataDescription = check_instance(
+            data_description, OutputVectorDataDescription
+        )
+
+        in_layer = tf.keras.layers.Input(input_dd.get_input_tensor_shape())
+
+        model = Sequential([
+            in_layer,
+
+            layers.Conv2D(16, 3, padding='same', activation=self.activation_function),
+            layers.MaxPooling2D(),
+
+            layers.Conv2D(32, 3, padding='same', activation=self.activation_function),
+            layers.MaxPooling2D(),
+
+            layers.Conv2D(64, 3, padding='same', activation=self.activation_function),
+            layers.MaxPooling2D(),
+
+            layers.Dropout(0.2),
+
+            layers.Flatten(),
+
+            layers.Dense(128, activation='relu', kernel_regularizer=regularizers.l2(self.regulation_value)),
+            layers.Dense(output_dd.get_output_size(), name="outputs", activation=self.final_activation)
+        ])
+        model.summary()
+        return model
+
+```
+In the next step, we will integrate this model in the training operation configuration.
```

Comment:
This has to be generalized, as it is not part of a tutorial anymore.

Reply:
done